packages/contracts/contracts/test/plugin-mock/UUPSUpgradeable/PluginUUPSUpgradeableSetupMock.sol

```diff
@@ -0,0 +1,187 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.10;
+
+import {PermissionLib} from "../../../core/permission/PermissionLib.sol";
+import {PluginSetup} from "../../../plugin/PluginSetup.sol";
+import {IPluginSetup} from "../../../plugin/IPluginSetup.sol";
+import {PluginUUPSUpgradeableV1Mock, PluginUUPSUpgradeableV2Mock, PluginUUPSUpgradeableV3Mock} from "./PluginUUPSUpgradeableMock.sol";
+import {mockPermissions, mockHelpers, mockPluginProxy} from "../PluginMockData.sol";
+
+contract PluginUUPSUpgradeableSetupV1Mock is PluginSetup {
+    address internal pluginBase;
+
+    constructor() {
+        pluginBase = address(new PluginUUPSUpgradeableV1Mock());
+    }
+
+    /// @inheritdoc IPluginSetup
+    function prepareInstallationDataABI() external view virtual override returns (string memory) {
+        return "";
+    }
+
+    /// @inheritdoc IPluginSetup
+    function prepareInstallation(address _dao, bytes memory)
+        public
+        virtual
+        override
+        returns (
+            address plugin,
+            address[] memory helpers,
+            PermissionLib.ItemMultiTarget[] memory permissions
+        )
+    {
+        plugin = mockPluginProxy(pluginBase, _dao);
+        helpers = mockHelpers(1);
+        permissions = mockPermissions(0, 1, PermissionLib.Operation.Grant);
+    }
+
+    /// @inheritdoc IPluginSetup
+    function prepareUninstallationDataABI() external view virtual override returns (string memory) {
+        return "";
+    }
+
+    /// @inheritdoc IPluginSetup
+    function prepareUninstallation(
+        address _dao,
+        address _plugin,
+        address[] calldata _currentHelpers,
+        bytes calldata
+    ) external virtual override returns (PermissionLib.ItemMultiTarget[] memory permissions) {
+        (_dao, _plugin, _currentHelpers);
+        permissions = mockPermissions(0, 1, PermissionLib.Operation.Revoke);
+    }
+
+    /// @inheritdoc IPluginSetup
+    function getImplementationAddress() external view virtual override returns (address) {
+        return address(pluginBase);
+    }
+}
+
+contract PluginUUPSUpgradeableSetupV1MockBad is PluginUUPSUpgradeableSetupV1Mock {
+    function prepareInstallation(address _dao, bytes memory)
+        public
+        pure
+        override
+        returns (
+            address plugin,
+            address[] memory helpers,
+            PermissionLib.ItemMultiTarget[] memory permissions
+        )
+    {
+        (_dao);
+        plugin = address(0); // The bad behaviour is returning the same address over and over again
+        helpers = mockHelpers(1);
+        permissions = mockPermissions(0, 1, PermissionLib.Operation.Grant);
+    }
+}
+
+contract PluginUUPSUpgradeableSetupV2Mock is PluginUUPSUpgradeableSetupV1Mock {
```

Comment:
If we create all mocks in one file, do we need the folder?
I thought the reason we have the folder (/Cloneable or /UUPSUpgradable) is to have each contract separate

Reply:
I agree but I would wait for https://github.com/aragon/core/pull/127. Otherwise, we have too many merge conflicts